tests/backup/backup_restore_basic_test.go

```diff
@@ -895,3 +896,197 @@ var _ = Describe("{CustomResourceRestore}", func() {
+				backupNames = append(backupNames, backupName)
+			}
+		})
+
+		Step("Check if all backups are synced or not", func() {
+			log.InfoD("Check if backups created before are synced or not")
+
+			// Wait for backups to get synced
+			for i := 0; i < 1000; i++ {
+				fetchedBackupNames, _ := GetAllBackupsAdmin()
+				log.Infof(fmt.Sprintf("Fetched backups %d", len(fetchedBackupNames)))
+				if len(fetchedBackupNames) == len(backupNames) {
+					break
+				} else {
+					time.Sleep(30)
+				}
+			}
+			fetchedBackupNames, err := GetAllBackupsAdmin()
+			log.FailOnError(err, "Getting a list of all backups")
+			dash.VerifyFatal(len(fetchedBackupNames), len(backupNames), "Comparing the expected and actual number of backups")
+			var bkp *api.BackupObject
+			backupDriver := Inst().Backup
+			bkpEnumerateReq := &api.BackupEnumerateRequest{
+				OrgId: orgID}
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx failed")
+			curBackups, err := backupDriver.EnumerateBackup(ctx, bkpEnumerateReq)
+			for _, bkp = range curBackups.GetBackups() {
+				backupInspectRequest := &api.BackupInspectRequest{
+					Name:  bkp.Name,
+					Uid:   bkp.Uid,
+					OrgId: orgID,
+				}
+				resp, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+				log.FailOnError(err, "Inspect each backup from list")
+				actual := resp.GetBackup().GetStatus().Status
+				expected := api.BackupInfo_StatusInfo_Success
+				dash.VerifyFatal(actual, expected, "Check each backup for success status")
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(contexts)
+		log.InfoD("Deleting the deployed apps after the testcase")
+		for i := 0; i < len(contexts); i++ {
+			opts := make(map[string]bool)
+			opts[SkipClusterScopedObjects] = true
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			err := Inst().S.Destroy(contexts[i], opts)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Verify destroying app %s, Err: %v", taskName, err))
+		}
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+
+		log.Infof("Deleting registered clusters for admin context")
+		err = DeleteCluster(SourceClusterName, orgID, ctx)
+		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", SourceClusterName))
+		err = DeleteCluster(destinationClusterName, orgID, ctx)
+		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", destinationClusterName))
+
+		log.Infof("Cleaning up backup location - %s", customBackupLocationName)
+		err = DeleteBackupLocation(customBackupLocationName, backupLocationUID, orgID, true)
+		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", customBackupLocationName))
+		log.Infof("Cleaning cloud credential")
+		//TODO: Eliminate time.Sleep
+		time.Sleep(time.Minute * 5)
+		err = DeleteCloudCredential(credName, orgID, cloudCredUID)
+		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cloud cred %s", credName))
```

Comment:
Please use `CleanupCloudSettingsAndClusters` function which does all the required cleanup

Reply:
Done